pkg/trace/agent/agent.go

```diff
@@ -112,6 +111,9 @@ func (a *Agent) Run() {
 	info.UpdatePreSampler(*a.Receiver.PreSampler.Stats()) // avoid exposing 0
 
 	a.start()
+	for i := 0; i < runtime.NumCPU()*2; i++ {
```

Comment:
I would say to make it configurable as we discussed. It means having a config option with `1` as default. That way we can release this patch in 6.12.0 without changing the behavior for everyone, while we continue to test it deeply before making the default a higher value in 6.13.0.

@gbbr makes sense?

Reply:
I think this value should stay directly related to the number of CPUs and should not be 1 by default. If we need to discuss this more, then maybe we should remove the last commit from this PR and have it as a separate pull request.

Reply:
I will move this change to a new PR, along with your comment. I think it's a separate matter deserving it's own discussion, especially if we want to hide this behind some sort of feature flag.

Reply:
Done: https://github.com/DataDog/datadog-agent/pull/3354